environments/hca-prod/terraform/buckets.tf

```
@@ -234,7 +247,14 @@ resource google_service_account_iam_binding hca_workload_identity_binding {
   service_account_id = module.hca_argo_runner_account.id
   role               = "roles/iam.workloadIdentityUser"
   members            = ["serviceAccount:${local.prod_project_id}.svc.id.goog[hca/argo-runner]"]
-  depends_on         = [module.hca_argo_runner_account]
+}
+
+resource google_service_account_iam_binding hca_dagster_workload_identity_binding {
+  provider = google-beta.target
+
+  service_account_id = module.hca_dagster_runner_account.id
+  role               = "roles/iam.workloadIdentityUser"
+  members            = ["serviceAccount:${data.google_project.current_project.name}.svc.id.goog[dagster/monster-dagster]"]
```

Unfortunately this project's id and name don't match, and through some awful debugging last year I learned that this particular thing needs to point to the project id, not the project name
```suggestion
  members            = ["serviceAccount:${local.prod_project_id}.svc.id.goog[dagster/monster-dagster]"]
```